pcsx2/PINE.cpp

```diff
@@ -646,7 +646,11 @@ PINEServer::IPCBuffer PINEServer::ParseCommand(std::span<u8> buf, std::vector<u8
 					goto error;
 				if (!SafetyChecks(buf_cnt, 1, ret_cnt, 0, buf_size)) [[unlikely]]
 					goto error;
-				Host::RunOnCPUThread([slot = FromSpan<u8>(buf, buf_cnt)] { VMManager::SaveStateToSlot(slot); });
+				Host::RunOnCPUThread([slot = FromSpan<u8>(buf, buf_cnt)] {
+					VMManager::SaveStateToSlot(slot, true, [slot](std::string error) {
```

Comment:
`error` can be a const reference.

Reply:
Little update:

When I found that broken zip_close call, I started looking more into our usage of libzip, and concluded that a lot of our error handling logic is broken. I've started working on refactoring our ZIP handling code to use RAII wrapper classes for libzip, this work is on the [zip_refactor](https://github.com/chaoticgd/pcsx2/tree/zip_refactor) branch.

That said, those changes are proving to be quite substantial, so for this PR I've just fixed the immediate problem, and will have to submit the ZIP refactor as a separate PR. Maybe we could leave it until 2.8.

You can retest now.